cylc/flow/task_events_mgr.py

```diff
@@ -284,6 +286,28 @@ def process_events(self, schd_ctx):
             elif ctx.ctx_type == self.HANDLER_JOB_LOGS_RETRIEVE:
                 self._process_job_logs_retrieval(schd_ctx, ctx, id_keys)
 
+    def _spawn_children(self, itask, output):
+        """Spawn children of output, or all children on succeed/fail"""
+        try:
+            children = itask.children[output]
+        except KeyError:
+            # No children depend on this ouput
+            children = []
```

Comment:
Should this go into an `else` block under the `if` below as the following block redefines `children`.

Reply:
Fair call.